manage/config-dialog.js

```diff
@@ -286,7 +313,34 @@ function configDialog(style) {
   }
 
   function updateVarOnChange() {
-    this.va.value = this.type !== 'checkbox' ? this.value : this.checked ? '1' : '0';
+    if (this.type === 'range') {
+      this.va.value = Number(this.value);
+      $('.current-value', this.closest('.config-range')).textContent = this.va.value + (this.va.units || '');
+    } else {
+      this.va.value = this.type !== 'checkbox' ? this.value : this.checked ? '1' : '0';
```

Comment:
For `number`, you can update `this.va.value` on change event, but update `input.value` when blurred:
```diff
diff --git a/manage/config-dialog.js b/manage/config-dialog.js
index 34559c4..23431bc 100644
--- a/manage/config-dialog.js
+++ b/manage/config-dialog.js
@@ -320,6 +320,9 @@ function configDialog(style) {
     if (this.type === 'range') {
       this.va.value = Number(this.value);
       $('.current-value', this.closest('.config-range')).textContent = this.va.value + (this.va.units || '');
+    } else if (this.type === "number") {
+      // assign a valid number to va.value
+      this.va.value = clampValue(Number(this.value), this.va);
     } else {
       this.va.value = this.type !== 'checkbox' ? this.value : this.checked ? '1' : '0';
     }
@@ -327,7 +330,8 @@ function configDialog(style) {
 
   // Applied to input[type=number]
   function updateVarOnBlur() {
-    this.value = this.va.value = clampValue(this.value, this.va);
+    // update the value of the input
+    this.value = this.va.value;
   }
 
   // Clamp input[type=number] to a valid range
```

Reply:
Good idea! But, maybe we shouldn't change the `updateVarOnBlur()` to ensure the values are the same.

Reply:
No longer necessary... the value is no longer saved if invalid.